### PR TITLE
Chore: Core Components

### PR DESCRIPTION
**Pull request by jazztinecruz**
- [x] Language
- [x] Typography
- [x] Icon
- [x] User
- [x] Added font-headliner
- [x] Added radius-base

### CODE REVIEW

File: core/components/icon/index.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import cn from "@utilities/cn";
+import React from "react";
+
+type IconProps = {
+  Svg: React.ForwardRefExoticComponent<
```
Comment (nikkoabucejo) [inline core/components/icon/index.tsx:5]: Can we rename this to `Element`

File: core/components/language/index.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"use client";
```
Comment (nikkoabucejo) [inline core/components/language/index.tsx]: This is suppose to be server component

File: core/components/button/index.tsx
Comment (nikkoabucejo) [inline core/components/button/index.tsx:1]: I don't we need button component anymore

File: core/components/typography/index.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+"use client";
```
Comment (nikkoabucejo) [inline core/components/typography/index.tsx:1]: Make this as server component

File: core/components/typography/index.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+import cn from "@utilities/cn";
+import type { ReactNode } from "react";
+
+type HeadingSize = "body" | "title" | "headliner" | "subtitle";
+
+type Props = {
+  heading?: HeadingSize;
+  children: ReactNode;
```
Comment (nikkoabucejo) [inline core/components/typography/index.tsx:10]: We have type `Children` globally

File: core/components/language/index.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+import { Chip } from "@nextui-org/react";
+import cn from "@utilities/cn";
+import type { ReactNode } from "react";
+
+type Props = {
+  children: ReactNode;
```
Comment (nikkoabucejo) [inline core/components/language/index.tsx:5]: We have Children type globally

File: core/components/typography/index.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  className?: string;
+};
+
+const Typography = ({ heading = "body", children, className = "" }: Props) => {
```
Comment (nikkoabucejo) [inline core/components/typography/index.tsx:14]: Remove the `Body` this will cause errors on Hydration, I think we should default span tag

**Review by nikkoabucejo — CHANGES_REQUESTED** [review]
Please see comments

File: core/components/typography/index.tsx (hunk 6 above)
Comment (jazztinecruz) [reply core/components/typography/index.tsx:14]: I renamed the Body to Base. but I think p tag is the right default.

File: core/components/language/index.tsx (hunk 5 above)
Comment (jazztinecruz) [reply core/components/language/index.tsx:5]: Sure. I didnt see that.

File: core/components/typography/index.tsx (hunk 3 above)
Comment (jazztinecruz) [reply core/components/typography/index.tsx:1]: Done!

File: core/components/button/index.tsx
Comment (jazztinecruz) [reply core/components/button/index.tsx:1]: Resolved

File: core/components/language/index.tsx (hunk 2 above)
Comment (jazztinecruz) [reply core/components/language/index.tsx]: Resolved!

File: core/components/icon/index.tsx (hunk 1 above)
Comment (jazztinecruz) [reply core/components/icon/index.tsx:5]: Done!

**Review by nikkoabucejo — APPROVED** [review]
LGTM

**Pull request merged**